allennlp/data/semparse/worlds/nlvr_world.py

```diff
@@ -92,17 +101,39 @@ class NlvrWorld(World):
         structured_rep from the JSON file.
     """
     # pylint: disable=too-many-public-methods
-    # TODO(pradeep): Define more spatial relationship methods: above, below, left_of, right_of..
+    # TODO(pradeep): Define more spatial relationship methods: left_of, right_of..
+    # They should be defined for objects within the same box.
     def __init__(self, world_representation: List[List[JsonDict]]) -> None:
         super(NlvrWorld, self).__init__(global_type_signatures=types.COMMON_TYPE_SIGNATURE,
                                         global_name_mapping=types.COMMON_NAME_MAPPING,
-                                        num_nested_lambdas=1)
+                                        num_nested_lambdas=0)
         self._boxes = set([Box(object_list, "box%d" % index)
                            for index, object_list in enumerate(world_representation)])
         self._objects: Set[Object] = set()
         for box in self._boxes:
             self._objects.update(box.objects)
 
+        self._number_operators = {"equals": operator.eq,
+                                  "not_equals": operator.ne,
+                                  "greater": operator.gt,
+                                  "lesser": operator.lt,
+                                  "greater_equals": operator.ge,
+                                  "lesser_equals": operator.le}
+
+        self._set_unary_operators = {"same": self._same,
+                                     "different": self._different}
+
+        self._set_binary_operators = {"all_equals": self._all_equals,
+                                      "any_equals": self._any_equals,
+                                      "none_equals": self._none_equals}
+
+        self._count_functions = {"count": self.count,  # type: ignore
```

Comment:
I am confused about why some of these functions have underscores and some of them don't.

Reply:
They should all have underscores. Fixed it.